utils/timed_store.go

```diff
@@ -81,6 +85,11 @@ func (self *TimedStore) Add(timestamp time.Time, item interface{}) {
 		self.buffer = self.buffer[index:]
 	}
 
+	// Remove any elements if over our max size.
+	if self.maxItems >= 0 && len(self.buffer) > self.maxItems {
```

Comment:
Yeah, I was playing around with that. It adds a bunch of complexity for relatively little gain, but I'll think about cleaning it up and sending a patch.